desktop/core/src/desktop/middleware.py

```diff
@@ -953,3 +954,17 @@ def __call__(self, request):
       request.META['HTTP_X_FORWARDED_FOR'] = request.META['REMOTE_ADDR']
 
     return self.get_response(request)
+
+
+class CacheControlMiddleware:
+  def __init__(self, get_response):
+    self.get_response = get_response
+    self.custom_cache_control = CUSTOM_CACHE_CONTROL.get()
+
+  def __call__(self, request):
+    response = self.get_response(request)
+    if self.custom_cache_control:
```

Comment:
Why not also unload the middleware if the config is false?

cf: https://docs.djangoproject.com/en/4.2/topics/http/middleware/#marking-middleware-as-unused

Example:
https://github.com/cloudera/hue/blob/876d88c95d03ca82cd7a95fcdd806720ec3fde74/desktop/core/src/desktop/middleware.py#L433-L435

Reply:
added changes to unload the middleware if the config is false